### PR TITLE
Status output for hubbot

### PR DESCRIPTION
**Pull request by dperpeet**

### CODE REVIEW

File: hubbot [hunk 1]
```diff
@@ -209,30 +231,177 @@ def consider_head(head, master):
...
+
+    <title>Cockpit - Hubbot Status</title>
+
+    <!-- Latest compiled and minified CSS -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css">
+
+    <!-- Optional theme -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap-theme.min.css">
+
+    <!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
```
Comment (dperpeet) [inline hubbot]: do we need this?

File: hubbot [hunk 2]
```diff
@@ -209,30 +231,177 @@ def consider_head(head, master):
...
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <!-- The above 3 meta tags *must* come first in the head; any other head content must come *after* these tags -->
+    <meta name="description" content="">
+    <meta name="author" content="Cockpit Project">
+    <!--<link rel="icon" href="../../favicon.ico">-->
+
+    <title>Cockpit - Hubbot Status</title>
+
+    <!-- Latest compiled and minified CSS -->
+    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap.min.css">
```
Comment (dperpeet) [inline hubbot]: do we have a cockpit bootstrap on our server that we should use?

Comment (mvollmer) [inline hubbot]: We can copy one into the right place.  But isn't the CDN version good enough?

File: hubbot (hunk 1 above)
Comment (mvollmer) [inline hubbot]: doesn't hurt, does it?

File: hubbot [hunk 3]
```diff
@@ -288,34 +442,37 @@ if "HUBBOT_NEW_STYLE" in os.environ and os.environ["HUBBOT_NEW_STYLE"]:
     new_style = True
     work = "%s/hubbot.new/" % os.environ['HOME']
     work_url = "http://files.cockpit-project.org/hubbot.new"
-    github_status_context = "hubbot/f22/x86_64/new"
+    github_status_context = "hubbot/%s/%s/new" % (default_short, default_arch.replace('_', '-'))
```
Comment (mvollmer) [inline hubbot]: I think this is wrong since the new style tests are not yet affected by default_os, etc.

**Comment by mvollmer** [comment]
`hubbot scan` fails for me like this:

```
Traceback (most recent call last):
  File "./hubbot", line 474, in <module>
    scan_pulls()
  File "./hubbot", line 404, in scan_pulls
    publish_status(extended_info)
  File "./hubbot", line 361, in publish_status
    tables = print_jobs_table('pending jobs', filter(lambda p: p[2] in [pull_status_pending, pull_status_running], extended_info))
  File "./hubbot", line 297, in print_jobs_table
    github_get("commits/%s/status" % head['sha'])['statuses'] or [ ])
  File "./hubbot", line 296, in <lambda>
    status = filter(lambda s: s['context'] == github_status_context and 'hubbot' in s['target_url'],
TypeError: argument of type 'NoneType' is not iterable
```

**Comment by mvollmer** [comment]
Potential fix:

``` 
         comment = job[3]
         # get status filtered by entries that concern us
-        status = filter(lambda s: s['context'] == github_status_context and 'hubbot' in s['target_url'],
+        status = filter(lambda s: s['context'] == github_status_context and s['target_url'],
                         github_get("commits/%s/status" % head['sha'])['statuses'])
         if status:
             # most recent entry will be first
```

**Comment by mvollmer** [comment]
Successful runs seem to sorted into the "pending" table. In fact, there is no `pull_status_success`.  Why?

**Comment by mvollmer** [comment]
> Successful runs seem to sorted into the "pending" table.

Right, they were in "pending" because "Hubbot was happy but will do this again because master has changed".  However, the status report didn't show that because it will only use "already passed, but master changed" when this condition has just now been detected, not when it has been detected and recorded in a previous run.

In any case, the point of this status page is not to replicate the github pull request page, is it?

**Comment by dperpeet** [comment]
> In any case, the point of this status page is not to replicate the github pull request page, is it?

I think they should match up. I'll see if it can be fixed without too much effort.

File: hubbot (hunk 2 above)
Comment (dperpeet) [inline hubbot]: leave it until it breaks :)

**Comment by dperpeet** [comment]
> I think they should match up. I'll see if it can be fixed without too much effort.

fixed: "was happy" is properly marked 
also, "needs rebase" is shown as detail if that was the fail reason

**Pull request merged**